metric/bigtable/src/main/java/com/spotify/heroic/metric/bigtable/api/BigtableMutatorImpl.java

```diff
@@ -166,7 +156,13 @@ public void onFailure(Throwable t) {
 
     private void flush() {
         synchronized (lock) {
-            tableToBulkMutation.values().stream().forEach(BulkMutation::flush);
+            tableToBulkMutation.values().stream().forEach(mutation -> {
+                try {
+                    mutation.flush();
```

Comment:
This didn't use to block, so the `flush` method it's part of didn't block either. We should not change that, since it might eat up threads that are used in other places.

It also blocks under the `BigtableMutatorImpl#lock` Lock, which is used to queue up new mutations and is in the direct write path.

To work around this, we can either use a nonblocking API (if available), or spin up another thread pool with queue to deal with this here.